components/infobox/wikis/pokemon/infobox_person_player_custom.lua

```diff
@@ -0,0 +1,99 @@
+---
+-- @Liquipedia
+-- wiki=pokemon
+-- page=Module:Infobox/Person/Player/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Player = require('Module:Infobox/Person')
+local String = require('Module:StringUtils')
+local Class = require('Module:Class')
+local GameAppearances = require('Module:GetGameAppearances')
+local TeamHistoryAuto = require('Module:TeamHistoryAuto')
+local Role = require('Module:Role')
+local Region = require('Module:Region')
+
+local Injector = require('Module:Infobox/Widget/Injector')
+local Cell = require('Module:Infobox/Widget/Cell')
+local Title = require('Module:Infobox/Widget/Title')
+local Center = require('Module:Infobox/Widget/Center')
+
+local _pagename = mw.title.getCurrentTitle().prefixedText
+local _role
+local _role2
+local _EMPTY_AUTO_HISTORY = '<table style="width:100%;text-align:left"></table>'
+
+local CustomPlayer = Class.new()
+
+local CustomInjector = Class.new(Injector)
+
+local _args
+
+function CustomPlayer.run(frame)
+	local player = Player(frame)
+	_args = player.args
+
+	player.adjustLPDB = CustomPlayer.adjustLPDB
+	player.createWidgetInjector = CustomPlayer.createWidgetInjector
+
+	return player:createInfobox(frame)
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'history' then
+		local manualHistory = _args.history
+		local automatedHistory = TeamHistoryAuto._results({
+			convertrole = 'true',
+			player = _pagename
+		}) or ''
+		automatedHistory = tostring(automatedHistory)
+		if automatedHistory == _EMPTY_AUTO_HISTORY then
+			automatedHistory = nil
+		end
+
+		if not (String.isEmpty(manualHistory) and String.isEmpty(automatedHistory)) then
```

Comment:
Currently not much of transfer that was added for Pokemon,a handful that likely existed only as just a TeamHistory entry in player pages. I think it might be ok as a safeguard for pages that fell under those (which eventually will have to migrate those TH entry only into actual transfer, even if its a nosource)

Reply:
I think what hjp is getting at is that the logic in that if statement is flawed